frontend/app/work_packages/services/work-package-field-service.js

```diff
@@ -53,6 +53,9 @@ module.exports = function(
     // TODO: extract to strategy if new cases arise
     if (field === 'date') {
       // nope
+      if(workPackage.links.children) {
+        return false;
+      }
```

Comment:
It should suffice to change the following line below

````JavaScript
var isWritable = schema.props[field].writable;
````

to 

````JavaScript
var isWritable = workPackage.form.embedded.schema.props[field].writable;
````

I can explain the disconnect between schema and form-schema:

The schema endpoint describes general properties of a work package. However, it is not completely specific to the work package that is linking it. This might be a weakness of the API, I am not quite sure yet as it was supposed to be a compromise between accuracy and efficiency (having a shared schema for many work packages).

However, the schema embedded in the form to edit a work package is always matching the current work package 100%. Thus special cases like "parents can't have their date written" are considered there.